File: packages/paso-py/usepaso/utils/color.py
"""
ANSI color utilities for CLI output.
Uses 24-bit true color (brand palette) when supported, falls back to standard ANSI.
Respects NO_COLOR env var (https://no-color.org/) and non-TTY stderr.

Brand colors from usepaso.dev/brand tokens.css:
  success: #4e916a (sage green)
  error:   #c5515c (muted red)
  warning: #c6813d (amber/rust)
  info:    #86bfcb (teal)
  muted:   #7a7773 (warm gray)
"""

import os
import sys


def _enabled():
    return 'NO_COLOR' not in os.environ and hasattr(sys.stderr, 'isatty') and sys.stderr.isatty()


def _true_color():
    if not _enabled():
        return False
    ct = os.environ.get('COLORTERM', '')
    tp = os.environ.get('TERM_PROGRAM', '')
    return ct in ('truecolor', '24bit') or tp in ('iTerm.app', 'vscode')


def _rgb(r: int, g: int, b: int, s: str) -> str:
    return f'\x1b[38;2;{r};{g};{b}m{s}\x1b[39m'


def _ansi(code: int, s: str) -> str:
    return f'\x1b[{code}m{s}\x1b[39m'


def green(s: str) -> str:
    """Brand: --color-success #4e916a"""
    if not _enabled():
        return s
    return _rgb(78, 145, 106, s) if _true_color() else _ansi(32, s)


def red(s: str) -> str:
    """Brand: --color-error #c5515c"""
    if not _enabled():
        return s
    return _rgb(197, 81, 92, s) if _true_color() else _ansi(31, s)


def yellow(s: str) -> str:
    """Brand: --color-warning #c6813d"""
    if not _enabled():
        return s
    return _rgb(198, 129, 61, s) if _true_color() else _ansi(33, s)


def cyan(s: str) -> str:
    """Brand: --teal-bright #86bfcb"""
    if not _enabled():
        return s
    return _rgb(134, 191, 203, s) if _true_color() else _ansi(36, s)


def dim(s: str) -> str:
    """Brand: --dark-text-muted #7a7773"""
    if not _enabled():
        return s
    if _true_color():
        return _rgb(122, 119, 115, s)
    return f'\x1b[2m{s}\x1b[22m'


def bold(s: str) -> str:
    if not _enabled():
        return s
    return f'\x1b[1m{s}\x1b[22m'

File: packages/paso-js/src/utils/color.ts
/**
 * ANSI color utilities for CLI output.
 * Uses 24-bit true color (brand palette) when supported, falls back to standard ANSI.
 * Respects NO_COLOR env var (https://no-color.org/) and non-TTY stderr.
 *
 * Brand colors from usepaso.dev/brand tokens.css:
 *   success: #4e916a (sage green)
 *   error:   #c5515c (muted red)
 *   warning: #c6813d (amber/rust)
 *   info:    #86bfcb (teal)
 *   muted:   #7a7773 (warm gray)
 */

const enabled = !process.env.NO_COLOR && process.stderr.isTTY !== false;

const trueColor =
  enabled &&
  (process.env.COLORTERM === 'truecolor' ||
    process.env.COLORTERM === '24bit' ||
    process.env.TERM_PROGRAM === 'iTerm.app' ||
    process.env.TERM_PROGRAM === 'vscode');

function rgb(r: number, g: number, b: number) {
  if (!enabled) return (s: string) => s;
  if (trueColor) return (s: string) => `\x1b[38;2;${r};${g};${b}m${s}\x1b[39m`;
  // Fall through to caller's ANSI fallback
  return null;
}

function ansi(open: number, close: number) {
  if (!enabled) return (s: string) => s;
  return (s: string) => `\x1b[${open}m${s}\x1b[${close}m`;
}

// Brand: --color-success #4e916a
export const green = rgb(78, 145, 106) ?? ansi(32, 39);

// Brand: --color-error #c5515c
export const red = rgb(197, 81, 92) ?? ansi(31, 39);

// Brand: --color-warning #c6813d
export const yellow = rgb(198, 129, 61) ?? ansi(33, 39);

// Brand: --teal-bright #86bfcb
export const cyan = rgb(134, 191, 203) ?? ansi(36, 39);

// Brand: --dark-text-muted #7a7773
export const dim = rgb(122, 119, 115) ?? ansi(2, 22);

export const bold = ansi(1, 22);
